presto-orc/src/main/java/com/facebook/presto/orc/reader/BooleanSelectiveStreamReader.java

```diff
@@ -224,14 +224,9 @@ else if (filter == null) {
 
                 int succeedingPositionsToFail = filter.getSucceedingPositionsToFail();
                 if (succeedingPositionsToFail > 0) {
-                    int positionsToSkip = 0;
-                    for (int j = 0; j < succeedingPositionsToFail; j++) {
```

Comment:
@yingsu00 It is not clear to me why this change is correct. It seems like it assumes that positions array doesn't have holes between positions[i] and positions[i + succeedingPositionsToFail] and I can't see why would that always be the case.

Reply:
@yingsu00 I think you can eliminate the loop while preserving correctness like so:

```
int positionsToSkip = positions[i + succeedingPositionsToFail] - positions[i];
skip(positionsToSkip);
streamPosition += positionsToSkip;
i += succeedingPositionsToFail;
```